ckanext/dgu/lib/helpers.py

```diff
@@ -1201,19 +1201,27 @@ def prep_group_edit_data(data):
         if key not in data:
             data[key] = value
 
+
 def top_level_init():
-    # Top level initialisation previously done in layout_base to make sure it
-    # is available to all sub-templates. This is a bit nasty, and I think we
-    # would be better off splitting these c.* things either into separate helpers
-    # or into our own BaseController. Perhaps. TODO.
+    '''These 'globals' are initialised by Genshi's layout_base only - not done
+    in Jinja, so for new templates use instead: is_an_official() or
+    groups_for_current_user().
+    '''
     c.groups = groups_for_current_user()
     c.is_an_official = bool(c.groups or is_sysadmin())
 
+
 def is_an_official():
-    return bool(c.groups or is_sysadmin())
+    if is_sysadmin():
+        return True
+    return bool(groups_for_current_user())
+
 
 def groups_for_current_user():
-    return c.userobj.get_groups(group_type='organization') if c.userobj else []
+    if c.groups == '':
```

Comment:
I think you got it the wrong way round, or am I missing something? If the
user has no groups, c.groups gets set to [] the first time. And on
subsequent calls to groups_for_current_user() if you have ```if not
c.groups``` it will call get_groups every time. Whereas the way I had it
```if c.groups == ''``` it will not run get_groups and correctly return []
again.

On 3 September 2015 at 10:26, Ross Jones <notifications@github.com> wrote:

> In ckanext/dgu/lib/helpers.py
> <https://github.com/datagovuk/ckanext-dgu/pull/156#discussion_r38627510>:
>
> >
> >  def groups_for_current_user():
> > -    return c.userobj.get_groups(group_type='organization') if c.userobj else []
> > +    if c.groups == '':
>
> Might be best as
>
> if not c.groups:
>
> because just below you possibly set it to [] which would mean calling
> multiple times in template would call get_groups each time.
>
> I'll fix.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datagovuk/ckanext-dgu/pull/156/files#r38627510>.
>